### ISSUE TITLE
Failed Cluster Creation due to Unable to Download Cilium when Admin Machine connected via Proxy

### ISSUE THREAD

**Issue by rimaulana** [aws/eks-anywhere#2009]
**What happened**:
Cluster creation failed due to  unable to generate cilium template from public ECR Cilium chart. This happened when admin machine doesn't have direct internet connectivity and have connection via proxy server. Sample of log from eksctl anywhere create cluster with verbose output

```
2022-05-04T16:58:49.819Z        V0      Installing networking on workload cluster
2022-05-04T16:58:49.820Z        V6      Executing command       {"cmd": "/usr/bin/docker exec -i -e HELM_EXPERIMENTAL_OCI=1 eksa_1651683247359434078 helm template oci://public.ecr.aws/isovalent/cilium --version 1.9.13-eksa.2 --insecure-skip-tls-verify --namespace kube-system -f -"}
2022-05-04T16:59:19.947Z        V4      Task finished   {"task_name": "workload-cluster-init", "duration": "2m28.851142484s"}
2022-05-04T16:59:19.947Z        V4      ----------------------------------
2022-05-04T16:59:19.947Z        V4      Task start      {"task_name": "collect-cluster-diagnostics"}
2022-05-04T16:59:19.947Z        V0      collecting cluster diagnostics
2022-05-04T16:59:19.947Z        V0      collecting management cluster diagnostics
2022-05-04T16:59:19.949Z        V3      bundle config written   {"path": "harbor/generated/bootstrap-cluster-2022-05-04T16:59:19Z-bundle.yaml"}
2022-05-04T16:59:19.949Z        V1      creating temporary namespace for diagnostic collector                                                                                                                              {"namespace": "eksa-diagnostics"}
```

You can see that cluster diagnostic are being collected soon after trying to install Cilium. At this point, the workload cluster is up however, all nodes are in NotReady state due to no CNI and output of kubectl get ds -n kube-system showing that there was no Cilium daemonset.

**What you expected to happen**:
Admin machine with connection to internet via proxy should be able to successfully create EKS anywhere cluster.

**How to reproduce it (as minimally and precisely as possible)**:
Create EKS Anywhere cluster from admin machine that has connection to the internet via proxy (no direct internet access) and make sure that the cluster definition has proxy configuration section that is working

```
spec:
   ...
   proxyConfiguration:
      httpProxy: http-proxy-ip:port
      httpsProxy: https-proxy-ip:port
      noProxy:
      - list of no proxy endpoints
```

**Anything else we need to know?**:

This is happened due to on later version of EKS anywhere binary, the Cilium manifest file are generated by contacting public ecr chart

```
2022-05-04T16:58:49.820Z        V6      Executing command       {"cmd": "/usr/bin/docker exec -i -e HELM_EXPERIMENTAL_OCI=1 eksa_1651683247359434078 helm template oci://public.ecr.aws/isovalent/cilium --version 1.9.13-eksa.2 --insecure-skip-tls-verify --namespace kube-system -f -"}
```

while older binary are executing from a generated manifest (output from older version)

```
2022-05-04T11:29:02.237-0400	V2	Pulling docker image	{"image": "public.ecr.aws/eks-anywhere/cli-tools:v0.1.0-eks-a-4"}
2022-05-04T11:29:02.237-0400	V6	Executing command	{"cmd": "/usr/bin/docker pull public.ecr.aws/eks-anywhere/cli-tools:v0.1.0-eks-a-4"}
....
2022-05-04T11:39:07.480-0400	V0	Installing networking on workload cluster
2022-05-04T11:39:07.938-0400	V6	Executing command	{"cmd": "/usr/bin/docker run --name eksa_kubectl_1651678747938948762 -i --network host -v /path/eksanywhere -w /path/eksanywhere -v /var/run/docker.sock:/var/run/docker.sock --entrypoint kubectl public.ecr.aws/eks-anywhere/cli-tools:v0.1.0-eks-a-4 apply -f - --kubeconfig harbor/harbor-eks-a-cluster.kubeconfig"}
```
Trying to execute the same helm command manually verified the issue with cli-tools container unable to reach public ecr endpoint

```
docker exec -i -e HELM_EXPERIMENTAL_OCI=1 eksa_1651683247359434078 helm template oci://public.ecr.aws/isovalent/cilium --version 1.9.13-eksa.2 --insecure-skip-tls-verify --namespace kube-system --debuginstall.go:178: [debug] Original chart version: "1.9.13-eksa.2"
Error: failed to do request: Head "https://public.ecr.aws/v2/isovalent/cilium/manifests/1.9.13-eksa.2": dial tcp 99.83.145.10:443: i/o timeout
helm.go:88: [debug] Head "https://public.ecr.aws/v2/isovalent/cilium/manifests/1.9.13-eksa.2": dial tcp 99.83.145.10:443: i/o timeout
failed to do request
github.com/containerd/containerd/remotes/docker.(*request).do
        github.com/containerd/containerd@v1.5.7/remotes/docker/resolver.go:569
github.com/containerd/containerd/remotes/docker.(*request).doWithRetries
        github.com/containerd/containerd@v1.5.7/remotes/docker/resolver.go:576
github.com/containerd/containerd/remotes/docker.(*dockerResolver).Resolve
        github.com/containerd/containerd@v1.5.7/remotes/docker/resolver.go:280
oras.land/oras-go/pkg/oras.Pull
        oras.land/oras-go@v0.4.0/pkg/oras/pull.go:45
helm.sh/helm/v3/internal/experimental/registry.(*Client).Pull
        helm.sh/helm/v3/internal/experimental/registry/client.go:266
helm.sh/helm/v3/pkg/getter.(*OCIGetter).get
        helm.sh/helm/v3/pkg/getter/ocigetter.go:57
helm.sh/helm/v3/pkg/getter.(*OCIGetter).Get
        helm.sh/helm/v3/pkg/getter/ocigetter.go:36
helm.sh/helm/v3/pkg/downloader.(*ChartDownloader).DownloadTo
        helm.sh/helm/v3/pkg/downloader/chart_downloader.go:99
helm.sh/helm/v3/pkg/action.(*ChartPathOptions).LocateChart
        helm.sh/helm/v3/pkg/action/install.go:754
main.runInstall
        helm.sh/helm/v3/cmd/helm/install.go:194
main.newTemplateCmd.func2
        helm.sh/helm/v3/cmd/helm/template.go:82
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371
```

The execution of the helm template generation should be provided with proxy, as the bootstrap node is also executed with proxy configuration. 

```
2022-05-04T16:54:24.408Z        V0      Creating new bootstrap cluster
2022-05-04T16:54:24.409Z        V4      Creating kind cluster   {"name": "harbor-eks-a-cluster", "kubeconfig": "harbor/generated/harbor.kind.kubeconfig"}
2022-05-04T16:54:24.409Z        V6      Executing command       {"cmd": "/usr/bin/docker exec -i -e HTTP_PROXY=10.80.61.20:3128 -e HTTPS_PROXY=10.80.61.20:3128 -e NO_PROXY=10.5.0.4,vcenter.sddc-34-205-199-90.vmwarevmc.com,10.0.0.0/8,192.168.0.0/16 eksa_1651683247359434078 kind create cluster --name harbor-eks-a-cluster --kubeconfig harbor/generated/harbor.kind.kubeconfig --image public.ecr.aws/eks-anywhere/kubernetes-sigs/kind/node:v1.21.9-eks-d-1-21-11-eks-a-8 --config harbor/generated/kind_tmp.yaml"}
```

Manually executing helm template command with proxy configuration shows a success

```
docker exec -i -e HELM_EXPERIMENTAL_OCI=1 -e HTTP_PROXY=10.80.61.20:3128 -e HTTPS_PROXY=10.80.61.20:3128 -e NO_PROXY=10.5.0.4,vcenter.sddc-34-205-199-90.vmwarevmc.com,10.0.0.0/8,192.168.0.0/16 eksa_1651683247359434078 helm template oci://public.ecr.aws/isovalent/cilium --version 1.9.13-eksa.2 --insecure-skip-tls-verify --namespace kube-system --debug
```

**Environment**:
- EKS Anywhere Release: ```v0.8.2```

**Comment by vivek-koppuru** [comment]
@krijan-devops Does this issue address the second issue with registry mirror offline mode that you were talking about?

https://github.com/aws/eks-anywhere/issues/1857

**Comment by vivek-koppuru** [comment]
This issue has been resolved as part of #2032 and will make it to the next release.

**Comment by vipulsha22** [comment]
Still facing same issue when creating a local cluster.
```
eksctl anywhere version
v0.9.1
```

```
2022-05-23T16:20:34.498+0530    V6      Executing command       {"cmd": "/usr/bin/docker exec -i -e NO_PROXY=localhost,192.168.0.0/16,10.96.0.0/12 -e HELM_EXPERIMENTAL_OCI=1 -e HTTP_PROXY=10.90.xxx.xxx:xxxx -e HTTPS_PROXY=10.90.xxx.xxx:xxxx eksa_1653302103979616229 helm template oci://public.ecr.aws/isovalent/cilium --version 1.9.13-eksa.2 --insecure-skip-tls-verify --namespace kube-system -f -"}
2022-05-23T16:20:52.252+0530    V9      docker  {"stderr": "Error: failed to download \"oci://public.ecr.aws/isovalent/cilium\" at version \"1.9.13-eksa.2\"\n"}
2022-05-23T16:20:52.252+0530    V4      Task finished   {"task_name": "workload-cluster-init", "duration": "3m49.66018933s"}

```

Running below command directly on Docker host using same proxy is working fine.
`helm template oci://public.ecr.aws/isovalent/cilium --version 1.9.13-eksa.2`
